web/instances/fixers.go

```diff
@@ -362,37 +362,41 @@ func serviceTriggersFixer(c echo.Context) error {
 			if service.TriggerOptions == "" {
 				continue
 			}
-			if service.TriggerID != "" {
-				_, err := jobsSystem.GetTrigger(inst, service.TriggerID)
-				if errors.Is(err, job.ErrNotFoundTrigger) {
-					triggerID, err := app.CreateServiceTrigger(inst, slug, service)
-					if err != nil {
-						return err
+			var recreate bool
+			if service.TriggerID == "" {
+				for _, trigger := range triggers {
+					infos := trigger.Infos()
+					if infos.Debounce != service.Debounce {
+						continue
+					}
+					opts := infos.Type + " " + infos.Arguments
+					if opts != service.TriggerOptions {
+						continue
+					}
+					var msg serviceMessage
+					if err := json.Unmarshal(infos.Message, &msg); err != nil {
+						continue
 					}
-					service.TriggerID = triggerID
+					if msg.Name != name {
+						continue
+					}
+					service.TriggerID = infos.TID
 					update = true
```

Comment:
Isn't a `break` missing here to stop looping when we've found the appropriate trigger ID to use?

Reply:
We could add it as an optimization, but it's not a big deal if we don't have it